webdriver/tests/interaction/element_click.py

```diff
@@ -0,0 +1,142 @@
+from tests.support.inline import inline
+
+def click(session, element):
+    return session.transport.send("POST", "session/{session_id}/element/{element_id}/click"
+                                  .format(session_id=session.session_id,
+                                          element_id=element.id))
+
+
+# 14.1 Element Click
+
+def test_no_browsing_context(session, create_window):
+    # 14.1 step 1
+    session.url = inline("<p>This is not an editable paragraph.")
+    element = session.find.css("p", all=False)
+
+    session.window_handle = create_window()
+    session.close()
+
+    response = click(session, element)
+    assert_error(response, "no such window")
+
+
+def test_element_file_upload_state(session):
+    # 14.1 Step 3
+    session.url = inline("<input type=file value=Federer>")
+
+    element = session.find.css("input", all=False)
+    response = click(session, element)
+    assert_error(response, "invalid argument")
+
+
+def test_scroll_into_element_view(session):
+    # 14.1 Step 4
+    session.url = inline("<input type=text value=Federer><div style= \"height: 200vh; width: 5000vh\">")
+
+    # Scroll to the bottom right of the page
+    session.execute_script("window.scrollTo(document.body.scrollWidth, document.body.scrollHeight);")
+    element = session.find.css("input", all=False)
+    # Clear and scroll back to the top of the page
+    response = click(session, element)
+    assert_success(response)
+
+    # Check if element cleared is scrolled into view
+    rect = session.execute_script("return document.getElementsByTagName(\"input\")[0].getBoundingClientRect()")
+
+    pageDict = {}
+
+    pageDict["innerHeight"] = session.execute_script("return window.innerHeight")
+    pageDict["innerWidth"] = session.execute_script("return window.innerWidth")
+    pageDict["pageXOffset"] = session.execute_script("return window.pageXOffset")
+    pageDict["pageYOffset"] = session.execute_script("return window.pageYOffset")
+
+    assert rect["top"] < (pageDict["innerHeight"] + pageDict["pageYOffset"]) and \
+           rect["left"] < (pageDict["innerWidth"] + pageDict["pageXOffset"]) and \
+           (rect["top"] + element.rect["height"]) > pageDict["pageYOffset"] and \
+           (rect["left"] + element.rect["width"]) > pageDict["pageXOffset"]
+
+# TODO
+# Any suggestions on implementation?
+# def test_element_not_interactable(session):
+#     # 14.1 Step 5
+#     assert_error(response, "element not interactable")
+
+def test_element_click_intercepted(session):
+    # 14.1 Step 6
+    session.url = inline("""<input type=button value=Roger style=\"position: absolute; left: 10px; top: 10px\">
+                            <div style=\"position: absolute; height: 100px; width: 100px; background: rgba(255,0,0,.5); left: 10px; top: 5px\"></div>""")
+
+    element = session.find.css("input", all=False)
+    response = click(session, element)
+    assert_error(response, "element not interactable")
+
+
```

Comment:
Please a test where the we have an element with `pointer-events: none` over another element to make sure that we let the click go through to that?

Reply:
added that test. It looks like Firefox is allowing the click when pointer-events is set to none.

Reply:
brilliant, finding bugs in other browsers are good.

Reply:
You were probably testing an old Firefox or didn’t set the `moz:webdriverClick` capability to true. Firefox does cater for `pointer-events: none`.

If you still find it does not, can you please file a bug?